Source/Core/VideoBackends/Software/CopyRegion.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <Common/MathUtil.h>
+
+#include <cmath>
+
+namespace SW
+{
+  // Modified from
+  // http://tech-algorithm.com/articles/nearest-neighbor-image-scaling/
+  template <typename T>
+  void copy_region(T* source, const MathUtil::Rectangle<int>& srcrect, T* destination, const MathUtil::Rectangle<int>& dstrect)
```

Comment:
Also, does the original author of the code you modified state the license they allow their code to be used/incorportated into? If not, you should likely contact and ask the author. If there's no clear indication of any form of licensing on the site itself, it can't be assumed to public domain, or compatible with any license.

Reply:
Thanks lioncash!  Yeah, I didn't ask the author.  His "about" page says the following:

```
most if not all articles will usually be presented with example implementation source code, copy-paste ready
```

I'll see if I can contact him but the last post was from 2009.

Reply:
I sort of hate to say this, but nearest neighbor is such an "obvious" resampling algorithm to me and a lot of people that I don't know if we need to credit authorship?

Reply:
Better question.....

Nearest Neighbor? That's going to look really fugly and it's not what the hardware uses at all. At the very least use bilinear resampling.

Reply:
Yeah, agree that we should be using a better implementation that more closely matches hardware but this was the fastest implementation for getting Software "done" and matching the other backends with y-scaling.  

From the way I understand it, very few users use the Software backend, so this could likely be a PR later on.  Admittedly, I wasn't very passionate about this backend which is why I only implemented the bare-bones implementation.